code/modules/events/holiday/vday.dm

```diff
@@ -0,0 +1,93 @@
+
+/obj/item/weapon/valentine
+	name = "valentine"
+	desc = "A Valentine's card! Wonder what it says..."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "sc_Ace of Hearts_syndicate" // shut up
+	var/message = "A generic message of love or whatever."
+	burn_state = FLAMMABLE
+	w_class = 1
+
+/obj/item/weapon/valentine/New()
+	..()
+	message = pick("Roses are red / Violets are good / One day while Andy...",
+	               "My love for you is like the singularity. It cannot be contained.",
+	               "Will you be my lusty xenomorph maid?",
+	               "We go together like the clown and the external airlock.",
+	               "Roses are red / Liches are wizards / I love you more than a whole squad of lizards.",
+	               "Be my valentine. Law 2.",
+	               "You must be a mime, because you leave me speechless.",
+	               "I love you like Ian loves the HoP.",
+	               "You're hotter than a plasma fire in toxins.",
+	               "Are you a rogue atmos tech? Because you're taking my breath away.",
+	               "Could I have all access... to your jumpsuit?",
+	               "Call me the CMO, because I'm here to inspect your johnson.",
+	               "Space is black. So is CO2. Be my Valentine. What the fuck is CO2?",
+	               "I'm not a changeling, but you make me extend my proboscis.",
+	               "I just can't get EI NATH of you.",
+	               "You must be a nuke op, because you make my heart explode.",
+	               "Roses are red / Botany's a farm / Not being my Valentine / causes human harm.",
+	               "I want you more than an assistant wants gloves.",
+	               "If I was a security officer, I'd brig you all shift.",
+	               "You know what you wanna do, right? You wanna put a banging donk in me.",
+	               "Are you the janitor? Because I think I've fallen for you.",
+	               "You're always valid to my heart.",
+	               "I'd risk the wrath of the gods to bwoink you.",
+	               "You look as beautiful now as the last time you were cloned.",
+	               )
+
+/obj/item/weapon/valentine/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/pen) || istype(W, /obj/item/toy/crayon))
+		var/recipient = stripped_input(user, "Who is receiving this valentine?", "To:", null , 20)
+		var/sender = stripped_input(user, "Who is sending this valentine?", "From:", null , 20)
+		if(recipient && sender)
+			name = "valentine - To: [recipient] From [sender]"
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/candyheart
+	name = "candy heart"
+	icon = 'icons/obj/holiday_misc.dmi'
+	icon_state = "candyheart"
+	desc = "A heart-shaped candy that reads: "
+	list_reagents = list("sugar" = 4)
+	junkiness = 5
+
+/obj/item/weapon/valentine/New()
+	..()
+	desc = pick("A heart-shaped candy that reads: HONK ME",
+                "A heart-shaped candy that reads: ERP",
```

Comment:
Most of these are uninspiring meme ones try and replace the ERP MEMES and LEWD with more discrete innuendos.